### DISCUSSION TITLE
How to run test in local docker?

### DISCUSSION THREAD

**Discussion by adrianomic** [mastodon/mastodon#17147] [Q&A] [answered]
I want to run test in fresh environment. So would like to run test in local docker.
What command should I use to do that?

Thanks

**Comment by ClearlyClaire** [comment, marked as answer]
I usually run tests as follows:
```sh
export RAILS_ENV=test
bundle install --with 'test development'
bundle exec rails db:reset
bundle exec rails db:setup
bundle exec rspec
```

**Comment by adrianomic** [comment]
Do you keep elasticsearch running before start ``bundle exec rspec`` ?

**Comment by ClearlyClaire** [comment]
I do not use ElasticSearch

**Comment by adrianomic** [comment]
If I do not run ElasticSearch, I am getting failed test 'Profile I can change my account'

``spec/features/profile_spec.rb``

If I run ElasticSearch, 4 tests are failed.

**Reply by ClearlyClaire** [reply]
Are you using `RAILS_ENV=test`? Tests should not be using ElasticSearch at all.

**Comment by adrianomic** [comment]
Yeah I use it

**Comment by adrianomic** [comment]
Thanks for your clear answer @ClearlyClaire
